metricbeat/module/mysql/status/status.go

```diff
@@ -104,3 +104,13 @@ func (m *MetricSet) loadStatus(db *sql.DB) (map[string]string, error) {
 
 	return mysqlStatus, nil
 }
+
+// Close closes the database connection and prevents future queries.
+func (m *MetricSet) Close() error {
+	err := m.db.Close()
```

Comment:
This should have a nil check for the `db`. And `errors.Wrap()` can safely be called when `err` is nil and it just returns `nil`.

```
if m.db == nil {
  return nil
}
return errors.Wrap(m.db.Close(), "failed to close mysql database client")
```

Reply:
The same fix should be added to `metricbeat/module/mysql/galera_status/status.go`.

Reply:
No, this path isn't reachable with a nil `db`.